sympy/functions/combinatorial/factorials.py

```diff
@@ -173,6 +173,60 @@ def eval(cls, n):
 
                     return Integer(result)
 
+    def _facmod(self, n, q):
+        res, N = 1, int(_sqrt(n))
+        m, pw = 2, [1]*N
```

Comment:
Could there be some comment explaining the purpose of `pw`?

BTW, on the next line, the upper limit of the range seems correct to me but in `factorint` the `+ 1` bit is missing.